### PR TITLE
Fix Service Depot Rally point path finding

### LINKED ISSUE #3757: Repair pad pathfinding
In all mods when I use the Repair pad and set a waypoint somewhere and the repaired vehicle finds a single blocking unit or building in its linear path it just simply stop and don't move anymore unless I give it a new order.

Perhaps it is related to the path finding algorithm which has also many issues. Some unit stop moving in a large group even it has enough space to do it.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/865206-repair-pad-pathfinding?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by rob-v**
Fix -Repaired unit moved towards Rally point but stopped on first obstacle instead of finding path like with Move order or with Production Rally Point.

Fixes #3757, #12965

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Repairable.cs
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Repairable.cs]: You can't hardcode Move here, that only works for Mobile, so will break this for other `IMove` implementations.

**Comment by rob-v** [comment]
Ok, thanks. silly mistake, I missed the fact that movement.MoveTo returns new Move so I just need to use the right one. I use the same as with Move order now and it works properly.

**Comment by pchote** [comment]
I don't think this is the best fix here - it looks like the real bug is inside `IMove.MoveTo(CPos, Actor)`, and while this might *technically* fix the service depot rally move, it doesn't resolve the real issue.

**Comment by rob-v** [comment]
That differently working Move ctor was suspicious to me, but as I don't have enough experience with pathing I didn't investigate more if that is intended behaviour or not. It looked to me also reasonable that move to rally point from SD works like Move order. I don't know if that ctor is buggy or correct but shouldn't be used here. FindPath(search) (not working) vs .FindUnitPath(mobile.ToCell, destination, self) (working). I will try to investigate if you don't agree with current fix.

**Comment by pchote** [comment]
This is caused by [Move.cs#L88](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Activities/Move/Move.cs#L88) running the pathfinder with `checkForBlocked = false`.  This makes it ignore *all* actors, not just the one we want.

**Comment by rob-v** [comment]
I don't know what is the right solution, if checkForBlocked = false is sometimes valid or we should change it to true (called from LayMines.Tick and twice from Repairable.AfterReachActivities).

**Comment by pchote** [comment]
It is valid for the nudge and scripted moves that are currently called via the `Move(Actor, CPos)` ctor, but should never be valid for the general moves that call `Move(Actor, CPos, Actor)`.

**Comment by rob-v** [comment]
Thanks, updated. The fix now looks trivial.

**Review by pchote — DISMISSED** [review]

**Comment by reaperrr** [comment]
As far as I can tell, this actually makes it worse.
If the rally point is blocked, the repaired actor now won't even leave the pad.

**Comment by rob-v** [comment]
You're right. Before this PR, it was blocked on a obstacle, but in this PR it doesn't leave SP in case of blocked rally point, though it finds a path otherwise.

Initially my PR resolved this issue by changing it to use the same Move(Actor self, CPos destination, WDist nearEnough) order used by normal Move order or move to RallyPoint of built units. We changed it to keep the current ctor but 'fix' one parameter.
I don't know what is the correct action - change it to use the same Move (which finds a path, isn't blocked by blocked rally point) or ...

**Comment by pchote** [comment]
`Move(Actor self, CPos destination, Actor ignoredActor)` should change to use the same kind of path searching as the others, but keep the `WithIgnoredActor` modifier.

**Comment by rob-v** [comment]
Updated.

**Comment by reaperrr** [comment]
Looks good to me now. If there's an obstacle in the middle of the path, the actor moves around it. If the rallypoint is blocked by a mobile actor, it nudges the blocker away. If the RP is blocked by an immobile actor (building), it moves as close as possible.
Code looks good, too, as far as I can tell, but I'd like @pchote to reconfirm his 👍 before this is merged, since he approved this before the code was updated.

**Review by reaperrr — APPROVED** [review]

File: OpenRA.Mods.Common/Activities/Move/Move.cs
Comment (pchote) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: Please set the default value for `ignoredActor` to null in the overload above instead of duplicating this.

File: OpenRA.Mods.Common/Traits/Mobile.cs
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: You can then omit this `null`.

File: OpenRA.Mods.Common/Traits/World/PathFinder.cs
Comment (pchote) [inline OpenRA.Mods.Common/Traits/World/PathFinder.cs]: Please use `ignoreActor` instead of `ignoredActor` for consistency with the rest of the code.

**Review by pchote — COMMENTED** [review]
Yup, looks good.  I have two minor style nits, then :+1:

File: OpenRA.Mods.Common/Traits/World/PathFinder.cs
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/World/PathFinder.cs]: Done. I wasn't sure about - There were few cases outside this class/PR with ignoredActor which I renamed also. There were 3 cases where not parameter but member name was IgnoredActor - renamed also. I kept WithIgnoredActor function name.

**Comment by rob-v** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d7753338e721d9969f2fbf54d1e52819a17c3488

**Pull request merged**
